### PR TITLE
Fix some Set-related performance regressions introduced by the new in…

### PR DESCRIPTION
**Pull request by swiftix**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [x] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

…dexing model.

Implement the FIXME and perform only one lookup instead of two.

rdar://26129485

### CODE REVIEW

**Comment by swiftix** [comment]
@swift-ci Please test

**Comment by swiftix** [comment]
@gribozavr This is a PR based on our discussion. Do you mind taking a look?

File: stdlib/public/core/HashedCollections.swift.gyb [hunk 1]
```diff
@@ -3843,6 +3858,63 @@ internal enum _Variant${Self}Storage<${TypeParametersDecl}> : _HashStorage {
...
+%if Self == 'Set':
+      return (inserted: false, memberAfterInsert: asNative.key(at: i.offset))
+%elif Self == 'Dictionary':
+      return (inserted: false, memberAfterInsert: asNative.value(at: i.offset))
+%end
+    }
+
+    let minCapacity = NativeStorage.minimumCapacity(
+                        minimumCount: asNative.count + 1,
+                        maxLoadFactorInverse: asNative.maxLoadFactorInverse)
```
Comment (gribozavr) [inline stdlib/public/core/HashedCollections.swift.gyb]: Please format this with 2 spaces on continuation line, the library style is different from LLVM style:

```
+let minCapacity = NativeStorage.minimumCapacity(
+  minimumCount: asNative.count + 1,
+  maxLoadFactorInverse: asNative.maxLoadFactorInverse)
```

**Comment by gribozavr** [comment]
@swiftix Thank you!  LGTM modulo the style comment.

File: stdlib/public/core/HashedCollections.swift.gyb [hunk 2]
```diff
@@ -3843,6 +3858,63 @@ internal enum _Variant${Self}Storage<${TypeParametersDecl}> : _HashStorage {
...
+    asNative.count += 1
+%elif Self == 'Dictionary':
+    asNative.initializeKey(key, value: value, at: i.offset)
+    asNative.count += 1
+%end
+
+    return (inserted: true, memberAfterInsert: value)
+  }
+
+
```
Comment (gribozavr) [inline stdlib/public/core/HashedCollections.swift.gyb]: Extra empty line.

**Pull request merged**